libminifi/src/core/json/JsonFlowSerializer.cpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "core/json/JsonFlowSerializer.h"
+
+#include "rapidjson/prettywriter.h"
+#include "utils/crypto/property_encryption/PropertyEncryptionUtils.h"
+
+#ifdef WIN32
+#pragma push_macro("GetObject")
+#undef GetObject  // windows.h #defines GetObject = GetObjectA or GetObjectW, which conflicts with rapidjson
+#endif
+
+namespace org::apache::nifi::minifi::core::json {
+
+void JsonFlowSerializer::encryptSensitiveProperties(rapidjson::Value &property_jsons, rapidjson::Document::AllocatorType &alloc,
+    const std::map<std::string, Property> &properties, const utils::crypto::EncryptionProvider &encryption_provider) const {
+  for (auto &property : property_jsons.GetObject()) {
+    std::string name{property.name.GetString()};
```

Comment:
we should also use `property.name.GetSize()` since it could contain a `\0` character (I see that it appends a trailing `\0` but I don't think we can rely on that as well)

Reply:
fixed in 52814c6cc538df8715d22ea213f6f5a355fc0118